pkg/actuators/client/client.go

```diff
@@ -20,86 +20,407 @@ import (
+	subnetID, err := c.GetSubnetIDbyName(subnetName, resourceGroupID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get Security Groups
+	securityGroups, err := c.GetSecurityGroupsByName(machineProviderConfig.PrimaryNetworkInterface.SecurityGroups, resourceGroupID, vpcID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create Instance Options
+	options := &vpcv1.CreateInstanceOptions{}
+
+	// Set Instance Prototype - Contains all the info necessary to provision an instance
+	options.SetInstancePrototype(&vpcv1.InstancePrototype{
+		Name: &machineName,
+		Image: &vpcv1.ImageIdentity{
+			ID: &imageID,
+		},
+		Profile: &vpcv1.InstanceProfileIdentity{
+			Name: &machineProviderConfig.Profile,
+		},
+		Zone: &vpcv1.ZoneIdentity{
+			Name: &machineProviderConfig.Zone,
+		},
+		ResourceGroup: &vpcv1.ResourceGroupIdentity{
+			ID: &resourceGroupID,
+		},
+		PrimaryNetworkInterface: &vpcv1.NetworkInterfacePrototype{
+			Subnet: &vpcv1.SubnetIdentity{
+				ID: &subnetID,
+			},
+			SecurityGroups: securityGroups,
+		},
+		VPC: &vpcv1.VPCIdentity{
+			ID: &vpcID,
+		},
+		UserData: &userData,
+	})
+
+	// Create a new Instance from an instance prototype object
+	instance, _, err := c.vpcService.CreateInstance(options)
+	if err != nil {
+		return nil, err
+	}
+
+	return instance, nil
+}
+
+// GetVPCIDByName Retrives VPC ID
+func (c *ibmCloudClient) GetVPCIDByName(vpcName string, resourceGroupID string) (string, error) {
+	// Initialize List Vpcs Options
+	vpcOptions := c.vpcService.NewListVpcsOptions()
+
+	// Set Resource Group ID
+	vpcOptions.SetResourceGroupID(resourceGroupID)
+
+	// Get a list all VPCs
+	vpcList, _, err := c.vpcService.ListVpcs(vpcOptions)
+	if err != nil {
+		return "", err
+	}
+
+	if vpcList != nil {
+		var vpcID string
+		for _, eachVPC := range vpcList.Vpcs {
+			if *eachVPC.Name == vpcName {
+				vpcID = *eachVPC.ID
+				return vpcID, nil
+			}
+		}
+	}
+
+	return "", fmt.Errorf("Could not retrieve VPC ID of name: %v", vpcName)
+}
+
+// GetCustomImageByName retrieves custom image from VPC by region and name
+func (c *ibmCloudClient) GetCustomImageByName(imageName string, resourceGroupID string) (string, error) {
+	// Initialize List Images Options
+	options := c.vpcService.NewListImagesOptions()
+
+	// Private images
+	options.SetVisibility(vpcv1.ImageVisibilityPrivateConst)
+	// Set Resource Group ID
+	options.SetResourceGroupID(resourceGroupID)
+	// Set Image name
+	options.SetName(imageName)
+
 	// List of all the private images in a region
 	privateImages, _, err := c.vpcService.ListImages(options)
 	if err != nil {
-		return nil, err
+		return "", err
+	}
+
+	if privateImages != nil {
+		// Update imageID when found a name match
+		var imageID string
+		for _, eachImage := range privateImages.Images {
+			if *eachImage.Name == imageName && *eachImage.Status == vpcv1.ImageStatusAvailableConst {
+				// Get Image ID
+				imageID = *eachImage.ID
+				return imageID, nil
```

Comment:
```suggestion
				return *eachImage.ID, nil
```
We probably don't need the `imageID` var